Stripe connected accounts can access earn regardless of plan

dsas opened the pull request:
#### Changes proposed in this Pull Request

The pages on earn to manage various earn settings are only available to
sites with a paid plan, even though payments will continue to be
processed and the stripe connection will still be in place.

This change allows free sites to use the Manage payments, Donations and
Premium Content so long as they have an existing stripe connection.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Make sure your sandbox is up to date
2. Use the generated calypso link below
3. Be on a paid plan
4. Go to wordpress.com/earn
5. Follow the prompts to link your stripe account as normal, if necessary
6. Go to wordpress.com/earn again, note that **Manage** Payments, Accept Donations and Tips and Manage your premium content all have CTAs  like Collect / Manage / Add respectively
7. Follow each of the three links, you should see the management screens
8. On the management screen press 'Payment plans' to modify the plans, you should be able to edit sites and add new ones. 
9. Remove your paid plan, e.g. using store admin
10. Reload wordpress.com/earn, you should see everything the same as in steps 5-7
11. Step 8 will be different - you should not see the add new button or the link to edit existing plans
12. Disconnect your stripe account
13. Go to wordpress.com/earn
14. You should see that the titles of the three cards have changed slightly, and that the wording on the CTAs has changed to 'unlock' 


** This is blocked from release until D78455-code ✅ and  897-gh-Automattic/wpcomsh are deployed **

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #59937

dsas committed:
Stripe connected accounts can access earn regardless of plan

The pages on earn to manage various earn settings are only available to
sites with a paid plan, even though payments will continue to be
processed and the stripe connection will still be in place.

This change allows free sites to use the Manage payments, Donations and
Premium Content so long as they have an existing stripe connection.

Fixes #59937

mmtr commented on client/my-sites/earn/home.tsx:
```diff
@@ -599,6 +601,7 @@ export default connect(
 			hasSetupAds: Boolean(
 				site?.options?.wordads || isRequestingWordAdsApprovalForSite( state, site )
 			),
+			stripeAvailable: hasConnectedAccount || ! isFreePlan,
```
Non-blocking: What do you think about using this opportunity to help the Limited Atomic pod and replace this plan check with a feature check? See p58i-cef-p2

It will require first to add the `memberships` feature to the `WPCOM_Features` class in `wpcom`, and then we could replace this check with something like

```suggestion
			stripeAvailable: hasConnectedAccount || hasActiveSiteFeature( state, siteId, FEATURE_MEMBERSHIPS ),
```

mmtr replied:
> It will require first to add the `memberships` feature to the `WPCOM_Features` class in `wpcom`

Scratch that. After a deeper look, we've realized we should use the `recurring-payments`, `donations` and `premium-content/container` features instead.

mmtr commented:
> This change allows free sites to use the Manage payments, Donations and
Premium Content so long as they have an existing stripe connection.

As discussed in our meetup, this is fine because the actual reason behind placing the Payment blocks under a paid plan is due to an anti-fraud security measurement. Sites with an expired plan already went through that anti-fraud check so it's fine they can still manage existing subscriptions.

Copons commented:
This works as expected, but I have a question/curiosity.

Should we prevent free sites to add new plans or edit existing ones?

<img width="1073" alt="Screenshot 2022-04-08 at 13 39 23" src="https://user-images.githubusercontent.com/2070010/162428744-0f0b2c62-3a0f-4939-a762-d3ed37e31286.png">

_/earn/payments-plans_

E.g. we could disable the "Add" button and "Edit" menu item, and we could add an "Unlock this feature" button near the Add one, linking to the plans purchase page.

dsas commented:
> Should we prevent free sites to add new plans or edit existing ones?

They can't actually use any of the new plans that they add, so doing this means they're aware sooner that they need to buy a new wpcom plan, which is nice. We should drive the decision by we decide for edit though.

Editing them is a bit of a coin toss, either choice is reasonable. It seems fair (and neater to my ocd brain) to prevent them from editing though so lets go with that.

dsas committed:
Update feature lists for payment blocks

Add features for donations and recurring payments.

Add a new FEATURE_PREMIUM_CONTENT_CONTAINER feature, which is the same
as in wpcom, and mark the previous one as legacy.

dsas committed:
Use feature based checks rather than plan checks

dsas committed:
Change availability of newsletter management

Like other earn blocks backed by stripe, allow newsletters to be managed
(or in this case - linked to the support doc) if the user has a stripe
connection, regardless of plan.

dsas committed:
Prevent people without feature access from modifying plans

Don't allow users with free stripe plan to add or edit plans - they can
remove them and look at them, but to add or modify them, they ought to
pay for a wpcom plan.

dsas commented:
> Should we prevent free sites to add new plans or edit existing ones?

I've removed them from the UI in this PR, I haven't made any back-end changes to prevent that.

mmtr reviewed (approved):
Nice work, thanks for replacing the plan checks with feature checks!
